ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/ontologies/views/OntologiesViewImpl.java

```diff
@@ -276,18 +300,14 @@ public void onSelection(SelectionEvent<Ontology> event) {
         return ontologySimpleComboBox;
     }
 
-    void updateButtonStatus() {
+    public void updateButtonStatus() {
         publishButton.setEnabled(selectedOntology != null && selectedOntology != activeOntology);
         saveHierarchy.setEnabled(selectedOntology != null);
+        deleteButton.setEnabled(selectedOntology != null && selectedOntology != activeOntology);
+        deleteHierarchy.setEnabled(selectedOntology != null && treeStore.getRootItems().size() > 0);
```

Comment:
Currently, once the admin marks an ontology as deleted, there's no way for them to get it back.
If they delete a hierarchy, they can just re-save the hierarchy's root and get back any classes that were deleted.